### PR TITLE
Fix the constructor called with no parameters add key-check for `getODataType()`

### PR DESCRIPTION
**Pull request by SilasKenneth**
## Summary

- Fixes the issue with missing key check before actually trying to get the key from the property dictionary for the `getODataType()` method.
- Fixes the issue with missing parameter for ctor in some classes which are actually called passing an argument.

## Generated code differences

Screenshot 2021-07-07 at 10 39 55

## Command line arguments to run these changes
N/A

## Links to issues or work items this PR addresses
Issue [#287](https://github.com/microsoftgraph/msgraph-sdk-php/issues/287)
PR [#539](https://github.com/microsoftgraph/msgraph-sdk-php/pull/539)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/556)

### CODE REVIEW

File: Templates/templates/PHP/Model/ComplexType.php.tt [hunk 1]
```diff
@@ -41,9 +41,11 @@ if (complex.IsBaseAbstractAndReferencedAsPropertyType() && !complex.IsAbstract)
 #>    /**
     * Set the @odata.type since this type is immediately descended from an abstract
     * type that is referenced as the type in an entity.
+    * @param array $propDict The property dictionary
     */
-    public function __construct()
+    public function __construct($propDict = array())
```
Comment (baywet) [inline Templates/templates/PHP/Model/ComplexType.php.tt:46]: adding parameters with default values to existing methods/ctors in dotnet is compatible from a prototype standpoint but constitutes a binary breaking change. I just want to double check this is not also the case in PhP.
https://docs.microsoft.com/en-us/dotnet/core/compatibility/
https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/breaking-changes#binary-breaking-change

Comment (SilasKenneth) [reply Templates/templates/PHP/Model/ComplexType.php.tt:46]: I would argue that the code wasn't working in the first place and adding this wouldn't require any code changes on the end of the client to make the new code work.

Comment (baywet) [reply Templates/templates/PHP/Model/ComplexType.php.tt:46]: thanks for confirming

**Review by baywet — APPROVED** [review]

**Review by MIchaelMainer — DISMISSED** [review]

File: Templates/templates/PHP/Model/EntityType.php.tt [hunk 2]
```diff
@@ -257,7 +257,10 @@ foreach(var property in entity.Properties.Where(prop => prop.Type.GetTypeString(
     */
```
Comment (Ndiritu) [inline Templates/templates/PHP/Model/EntityType.php.tt:257]: `@return string|null`

**Review by Ndiritu — CHANGES_REQUESTED** [review]

**Review by Ndiritu — APPROVED** [review]

**Comment by SilasKenneth** [comment]
Hi @baywet @MIchaelMainer , kindly approve this, I made a change that dismissed your review.

**Review by baywet — APPROVED** [review]

**Pull request merged**